compiler/rustc_borrowck/src/polonius/typeck_constraints.rs

```diff
@@ -35,7 +41,191 @@ pub(super) fn convert_typeck_constraints<'tcx>(
                 }
             }
 
-            _ => {}
+            Locations::Single(location) => {
+                // This constraint is marked as holding at one location, we localize it to that
+                // location or its successor, depending on the corresponding MIR
+                // statement/terminator. Unfortunately, they all show up from typeck as coming "on
+                // entry", so for now we modify them to take effects that should apply "on exit"
+                // into account.
+                //
+                // FIXME: this approach is subtle, complicated, and hard to test, so we should track
+                // this information better in MIR typeck instead, for example with a new `Locations`
+                // variant that contains which node is crossing over between entry and exit.
+                let point = liveness.point_from_location(location);
+                let (from, to) = if let Some(stmt) =
+                    body[location.block].statements.get(location.statement_index)
+                {
+                    localize_statement_constraint(
+                        tcx,
+                        body,
+                        stmt,
+                        liveness,
+                        &outlives_constraint,
+                        location,
+                        point,
+                        universal_regions,
+                    )
+                } else {
+                    assert_eq!(location.statement_index, body[location.block].statements.len());
+                    let terminator = body[location.block].terminator();
+                    localize_terminator_constraint(
+                        tcx,
+                        body,
+                        terminator,
+                        liveness,
+                        &outlives_constraint,
+                        point,
+                        universal_regions,
+                    )
+                };
+                localized_outlives_constraints.push(LocalizedOutlivesConstraint {
+                    source: outlives_constraint.sup,
+                    from,
+                    target: outlives_constraint.sub,
+                    to,
+                });
```

Comment:
I wonder if  readability would be better here if you moved the `LocalizedOutlivesConstraint` construction deeper (into `compute_constraint_direction`), instead of propagating up a `(PointIndex, PointIndex)`. It'll then be much more clear the matching of `sup`/`sub` to `from`/`to`.